warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -210,6 +214,42 @@ func (*AzureSynapse) IsEmpty(_ context.Context, _ model.Warehouse) (empty bool,
 	return
 }
 
+
+func (as *AzureSynapse) createStagingTable(ctx context.Context, tableName string, columnMap model.TableSchema) (string, error) {
+	stagingTableName := warehouseutils.StagingTableName(
+		provider,
+		tableName,
+		tableNameLimit,
+	)
+
+	varcharCols, err := as.getStringColumnsWithVariableLength(ctx, tableName)
+	if err != nil {
+		return "", fmt.Errorf("getting varchar columns info: %w", err)
+	}
+
+	columnDefs := lo.MapToSlice(columnMap, func(name, dataType string) string {
+		if dataType, ok := varcharCols[name]; ok {
+			return fmt.Sprintf(`"%s" %s(max)`, name, dataType)
+		}
+		return fmt.Sprintf(`"%s" %s`, name, rudderDataTypesMapToAzureSynapse[dataType])
```

Comment:
@achettyiitr Will this work for other columns. can we just map rudder types to synapse column and attempt to load data in staging table?